src/index.ts

```diff
@@ -1,34 +1,59 @@
 import defu from 'defu'
+import { ModuleThis } from '@nuxt/types/config/module'
 import { name, version } from '../package.json'
 import type { Options } from '@swc/core'
 import type { NuxtOptionsBuild } from '@nuxt/types/config/build'
 
-function swcModule () {
-  const { nuxt } = this
+const defaultTargets = {
+  server: { node: 12 },
+  client: { ie: 9 },
+  modern: {
+    chrome: 60,
+    firefox: 54
+  }
+}
 
-  const swcOptions: Options = defu(nuxt.options.build.swc, {
-    // sync: true,
-    sourceMaps: false,
-    jsc: {
-      parser: {
-        dynamicImport: true
-      }
-    }
-  } as Options)
+function swcModule (this: ModuleThis) {
+  this.options.extensions.push('ts')
+  this.options.build.additionalExtensions = ['ts', 'tsx']
 
-  const swcTSOptions = defu(swcOptions, {
-    jsc: {
-      parser: {
-        syntax: 'typescript'
-      }
+  // Auto detect corejs version:
+  // https://github.com/nuxt/nuxt.js/tree/dev/packages/webpack/src/config/base.js#L124-L132
+  let coreJs: string
+  if (this.options.build.corejs === 'auto') {
+    try {
+      coreJs = require('core-js/package.json').version.split('.')[0]
+    } catch (_err) {
+      coreJs = '2'
     }
-  } as Options)
-
-  nuxt.options.extensions.push('ts')
-  nuxt.options.build.additionalExtensions = ['ts', 'tsx']
+  } else {
+    coreJs = String(this.options.build.corejs)
+  }
 
-  nuxt.hook('webpack:config', (configs) => {
+  this.nuxt.hook('webpack:config', (configs) => {
```

Comment:
Can we please keep using `nuxt` variable extracted from this?

Reply:
sure. just needed to move to `this.options` for type hinting, and thought it was simpler.

Reply:
(I realise we will ultimately likely move to nuxt/nuxt.options when building this module with kit but it's a straightforward change to make at that point.)

Reply:
cf: https://github.com/nuxt/nuxt.js/pull/9773